examples/hrtim/adc-trigger.rs

```diff
@@ -0,0 +1,150 @@
+        gpio::{gpioa::PA8, gpioa::PA9, Alternate, GpioExt, AF13},
+        hrtim::control::HrControltExt,
+        hrtim::output::HrOutput,
+        hrtim::HrPwmAdvExt,
+        hrtim::{control::Adc13Trigger, Pscl4},
+        pwr::PwrExt,
+        rcc::{self, RccExt},
+        stm32::Peripherals,
+    };
+
+    info!("start");
+
+    let dp = Peripherals::take().unwrap();
+    let cp = cortex_m::Peripherals::take().expect("cannot take core peripherals");
+
+    // Set system frequency to 16MHz * 15/1/2 = 120MHz
+    // This would lead to HrTim running at 120MHz * 32 = 3.84...
+    info!("rcc");
+    let pwr = dp.PWR.constrain().freeze();
+    let mut rcc = dp.RCC.freeze(
+        rcc::Config::pll().pll_cfg(rcc::PllConfig {
+            mux: rcc::PLLSrc::HSI,
+            n: rcc::PllNMul::MUL_15,
+            m: rcc::PllMDiv::DIV_1,
+            r: Some(rcc::PllRDiv::DIV_2),
+
+            ..Default::default()
+        }),
+        pwr,
+    );
+
+    let mut delay = cp.SYST.delay(&rcc.clocks);
+
+    let dma::stream::StreamsTuple(dma1ch1, ..) = dp.DMA1.split(&rcc);
+    let config = DmaConfig::default()
+        .transfer_complete_interrupt(true)
+        .circular_buffer(true)
+        .memory_increment(true);
+
+    info!("Setup Gpio");
+    let gpioa = dp.GPIOA.split(&mut rcc);
+    let pa0 = gpioa.pa0.into_analog();
+
+    info!("Setup Adc1");
+    let mut adc = dp
+        .ADC1
+        .claim(ClockSource::SystemClock, &rcc, &mut delay, true);
+
+    adc.set_external_trigger((
+        adc::config::TriggerMode::RisingEdge,
+        ExternalTrigger12::Hrtim_adc_trg_1,
+    ));
+    adc.enable_temperature(&dp.ADC12_COMMON);
+    adc.set_continuous(Continuous::Discontinuous);
+    adc.reset_sequence();
+    adc.configure_channel(&pa0, Sequence::One, SampleTime::Cycles_640_5);
+    adc.configure_channel(&Temperature, Sequence::Two, SampleTime::Cycles_640_5);
+
+    info!("Setup DMA");
+    let first_buffer = cortex_m::singleton!(: [u16; 10] = [0; 10]).unwrap();
+
+    let mut transfer = dma1ch1.into_circ_peripheral_to_memory_transfer(
+        adc.enable_dma(AdcDma::Continuous),
+        &mut first_buffer[..],
+        config,
+    );
+
+    transfer.start(|adc| adc.start_conversion());
+
+    let pin_a: PA8<Alternate<AF13>> = gpioa.pa8.into_alternate();
+    let pin_b: PA9<Alternate<AF13>> = gpioa.pa9.into_alternate();
+
+    // ...with a prescaler of 4 this gives us a HrTimer with a tick rate of 960MHz
+    // With max the max period set, this would be 960MHz/2^16 ~= 15kHz...
+    let prescaler = Pscl4;
+
+    //        .               .
+    //        .  50%          .
+    //         ------          ------
+    //out1    |      |        |      |
+    //        |      |        |      |
+    // --------      ----------      --------
+    let (hr_control, ..) = dp
+        .HRTIM_COMMON
+        .hr_control(&mut rcc)
+        .enable_adc_trigger1_source(Adc13Trigger::TimACmp3)
+        .enable_adc_trigger1_source(Adc13Trigger::TimACmp4)
```

Comment:
If this enable method was instead defined on `HrPwmControl` then the variables `cr3` and `cr4` could be passed directly to that method without the need for the enum.

Reply:
@no111u3 what do you think about something like this d8d2b92965d822afcdd5c9ac41ea101c28fa41b5 to avoid enums? (Will probably update the examples at some point)

Reply:
Hmm, I think it better to use one work variant instead of any workless.